File: src/main/java/com/fengsheng/gm/addrobot.java
package com.fengsheng.gm;

import com.fengsheng.Game;
import com.fengsheng.Player;
import com.fengsheng.RobotPlayer;
import com.fengsheng.Statistics;

import java.util.Map;
import java.util.function.Function;

public class addrobot implements Function<Map<String, String>, String> {
    @Override
    public String apply(Map<String, String> form) {
        try {
            int count = form.containsKey("count") ? Integer.parseInt(form.get("count")) : 0;
            count = count > 0 ? count : 99;
            synchronized (Game.class) {
                Game g = Game.getInstance();
                for (int i = 0; i < count; i++) {
                    if (g.isStarted())
                        break;
                    Player robotPlayer = new RobotPlayer();
                    robotPlayer.setPlayerName(Player.randPlayerName());
                    robotPlayer.setGame(g);
                    robotPlayer.getGame().onPlayerJoinRoom(robotPlayer, Statistics.getInstance().getTotalPlayerGameCount());
                }
            }
            return "{\"msg\": \"success\"}";
        } catch (NumberFormatException | NullPointerException e) {
            return "{\"error\": \"invalid arguments\"}";
        }
    }
}

File: src/main/java/gm/addrobot.kt
package com.fengsheng.gm

import com.fengsheng.*
import java.util.function.Function

class addrobot : Function<Map<String?, String>, String> {
    override fun apply(form: Map<String?, String>): String {
        return try {
            var count = if (form.containsKey("count")) form["count"]!!.toInt() else 0
            count = if (count > 0) count else 99
            synchronized(Game::class.java) {
                val g: Game = Game.Companion.getInstance()
                for (i in 0 until count) {
                    if (g.isStarted) break
                    val robotPlayer: Player = RobotPlayer()
                    robotPlayer.playerName = Player.Companion.randPlayerName()
                    robotPlayer.game = g
                    robotPlayer.game.onPlayerJoinRoom(
                        robotPlayer,
                        Statistics.Companion.getInstance().getTotalPlayerGameCount()
                    )
                }
            }
            "{\"msg\": \"success\"}"
        } catch (e: NumberFormatException) {
            "{\"error\": \"invalid arguments\"}"
        } catch (e: NullPointerException) {
            "{\"error\": \"invalid arguments\"}"
        }
    }
}